modules/renter/files_test.go

```diff
@@ -59,6 +59,39 @@ func TestFileAvailable(t *testing.T) {
 	}
 }
 
+func TestFileRedundancy(t *testing.T) {
+	testNData := []int{1, 2, 10}
+	for _, nData := range testNData {
+		rsc, _ := NewRSCode(nData, 10)
+		f := &file{
+			size:        1000,
+			erasureCode: rsc,
+			pieceSize:   100,
+		}
+
+		r := f.redundancy()
+		if r != 0 {
+			t.Error("expected 0 redundancy, got", r)
+		}
+
+		testRedundancies := []uint64{1, 6, 10}
+		for _, testR := range testRedundancies {
+			var fc fileContract
+			for i := uint64(0); i < f.numChunks(); i++ {
+				for j := uint64(0); j < testR; j++ {
+					fc.Pieces = append(fc.Pieces, pieceData{Chunk: i, Piece: j})
+				}
```

Comment:
loop nesting this deep makes the test difficult to understand. There's probably a better way to approach this test.